alchemy/src/encrypt.ts

```diff
@@ -1,11 +1,114 @@
+import crypto from "node:crypto";
+
+const KEY_LEN = 32;
+const SCRYPT_N = 16384;
+const SCRYPT_R = 8;
+const SCRYPT_P = 1;
+const SALT_LEN = 16;
+const IV_LEN = 12;
+const ALGO = "aes-256-gcm";
+
+interface Encrypted {
+  version: "v1";
+  ciphertext: string; // base64
+  iv: string; // base64
+  salt: string; // base64
+  tag: string; // base64
+}
+
+export function encrypt(value: string, key: string): Promise<Encrypted> {
+  return scryptEncrypt(value, key);
+}
+
+export function decryptWithKey(
+  value: string | Encrypted,
+  key: string,
+): Promise<string> {
+  if (typeof value === "string") {
+    return libsodiumDecrypt(value, key);
+  }
+  return scryptDecrypt(value, key);
+}
```

Comment:
Maybe add something like this, for inline documentation of changes in the system 😄 

```suggestion
/**
 * @legacy Decrypts a value while preserving backward compatibility with
 * deprecated legacy ciphertexts.
 */
export function decryptWithKey(
  value: string | Encrypted,
  key: string,
): Promise<string> {
  if (typeof value === "string") {
    return libsodiumDecrypt(value, key);
  }
  return scryptDecrypt(value, key);
}
```